lib/components/Project.js

```diff
@@ -1,75 +1,118 @@
 'use strict';
 
 var path = require('path');
-var glob = require('glob-all');
 var _ = require('lodash');
 var PropertiesReader = require('properties-reader');
+
 var fs = require('fs');
 var os = require("os");
 
-var contracts = require('../../api/contracts');
+var Project = require('../../api/project');
 
 var EXTENSION_LENGTH = ".properties".length;
 var LANGUAGE_CODE_LENGTH = 4;
 
-function getLanguageCodeFromFilePath(path) {
-	return path.slice(-(EXTENSION_LENGTH + LANGUAGE_CODE_LENGTH + 1), -EXTENSION_LENGTH);
+function getLanguageCodeFromFilePath(filePath) {
+	return filePath.slice(-(EXTENSION_LENGTH + LANGUAGE_CODE_LENGTH + 1), -EXTENSION_LENGTH);
 }
 
-var Project = function () {
+function DefaultProject() {
 
 	var projectDir;
-	var assetsPatterns = ['*/src/main/resources/**/*.@(js|html|css)'];
-	var translationFilesPatterns = ['*/src/main/resources/translations/*.properties'];
-	var uiFilesPatterns = ['*/src/main/resources/*.@(jsfiles|modules)'];
 
-	return {
+	var fileCategories = {
+		translations: {
+			name: 'translations',
+			patterns: ['*/src/main/resources/translations/*.properties']
+		},
+		uifiles: {
+			name: 'uifiles',
+			patterns: ['*/src/main/resources/*.@(jsfiles|modules)']
+		},
+		assets: {
+			name: 'assets',
+			patterns: ['*/src/main/resources/**/*.@(js|html|css)']
+		}
+	};
+
+	var fileManager;
 
+	return {
 		configure: function (hub, configuration) {
 			projectDir = configuration.projectDir;
-			assetsPatterns = configuration.assetsPatterns || assetsPatterns;
-			translationFilesPatterns = configuration.translationFilesPatterns || translationFilesPatterns;
+			fileCategories.assets.patterns = configuration.assetsPatterns || fileCategories.assets.patterns;
+			fileCategories.translations.patterns = configuration.translationFilesPatterns || fileCategories.translations.patterns;
+			fileCategories.uifiles.patterns = configuration.uiFilesPatterns || fileCategories.uifiles.patterns;
+
+			hub.requireService({
+				contract: contracts.fileManager,
+				component: this,
+				bind: this.bindFileManager,
+				unbind: this.unbindFileManager
+			});
 
 			hub.provideService({
-				contract: contracts.project,
+				contract: Project,
 				component: this
 			});
+
+
+		},
+		bindFileManager: function (service) {
+			fileManager = service;
 		},
+		unbindFileManager: function () {
+			fileManager = null;
+		},
+
 		start: function () {
+			hub.subscribe(this, '/framework/started', function() {
```

Comment:
Does this means that we will not watch for files when add projects on the fly(not supported for now)?